miden-lib/build.rs

```diff
@@ -1,14 +1,63 @@
 use assembly::{LibraryNamespace, MaslLibrary, Version};
-use std::io;
+use std::{env, fs, fs::File, io, io::BufRead, io::BufReader, io::Write, path::Path};
 
 // CONSTANTS
 // ================================================================================================
-const ASM_DIR_PATH: &str = "./asm";
-const ASL_DIR_PATH: &str = "./assets";
+const ASL_DIR: &str = "assets";
+const ASM_DIR: &str = "asm";
 
 // PRE-PROCESSING
 // ================================================================================================
 
+/// Recursively copies `src` into `dst`.
+///
+/// This function will overwrite the existing files if re-executed.
+///
+/// Panics:
+/// - If any of the IO operation fails.
+fn copy_directory<T: AsRef<Path>, R: AsRef<Path>>(src: T, dst: R) {
+    let mut prefix = src.as_ref().canonicalize().unwrap();
+    prefix.pop(); // keep all the files inside the `asm` folder
+
+    let target_dir = dst.as_ref().join(ASM_DIR);
+    if !target_dir.exists() {
+        fs::create_dir(target_dir).unwrap();
+    }
+
+    let dst = dst.as_ref();
+    let mut todo = vec![src.as_ref().to_path_buf()];
+
+    while let Some(goal) = todo.pop() {
+        for entry in fs::read_dir(goal).unwrap() {
+            let path = entry.unwrap().path();
+            if path.is_dir() {
+                let src_dir = path.canonicalize().unwrap();
```

Comment:
I forgot to mention this before, but this will probably break the copy procedure if there are sym or hard links, I just don't expect that we are going to be using that on our git repos. I did this mostly to resolver any `./` or `../` in the path.

Reply:
Yep - I think this shouldn't be a problem.